### DISCUSSION TITLE
Error with @web3auth/single-factor-auth

### DISCUSSION THREAD

**Discussion by nisa10880** [Web3Auth/Web3Auth#1061] [Help]
Hello 

I have an error when trying to use "@web3auth/single-factor-auth": "^4.2.0-alpha.0". 
I am working with ReactJS 18 with ViteJS.

`
   ✘ [ERROR] No matching export in "node_modules/bowser/src/bowser.js" for import "ENGINE_MAP"
    node_modules/@toruslabs/customauth/dist/customauth.esm.js:16:17:
      16 │ import Bowser, { ENGINE_MAP } from 'bowser';
`

**Comment by Saviour1001** [comment]
Hey there, would recommend using the version ```^3.3.0``` for now. 
Use this [example](https://github.com/Web3Auth/examples/tree/main/web-single-factor-auth/react-evm-sfa-example) as a reference for creating your project.

**Reply by nisa10880** [reply]
I still have the issue with ^3.3.0

**Reply by AlexMaio** [reply]
Same here

**Comment by edwardchew97** [comment]
Having the same error here

**Comment by nisa10880** [comment]
@web3auth/single-factor-auth depends on @toruslabs/customauth. I think this file should be updated.
https://github.com/torusresearch/CustomAuth/blob/master/src/utils/helpers.ts

There is an import issue.

**Comment by AlexMaio** [comment]
Hey, I have the same problem on my side if you can fix it

**Comment by shahbaz17** [comment]
Hey @nisa10880 @AlexMaio @edwardchew97 

We have released a fix to this. Please update your packages to the latest and try again. Let us know if you're still facing the issue.

**Reply by nisa10880** [reply]
I still have this error 

✘ [ERROR] No matching export in "node_modules/bowser/src/bowser.js" for import "ENGINE_MAP"
    node_modules/@toruslabs/customauth/dist/customauth.esm.js:16:17:
      16 │ import Bowser, { ENGINE_MAP } from 'bowser';

As I said in my previous comment I think you need to replace ENGINE_MAP by Bowser.ENGINE_MAP and remove it from import.

https://github.com/torusresearch/CustomAuth/blob/master/src/utils/helpers.ts

**Reply by nisa10880** [reply]
@shahbaz17 Should I post an issue in torussearch repo ?

**Reply by shahbaz17** [reply]
Hey @nisa10880 

The issue is now fixed. You can check out https://github.com/Web3Auth/examples/tree/main/web-single-factor-auth/react-evm-sfa-example. 

Please update the web3auth packages to the latest.

When using SFA and Web3Auth Core SDK together, you would need to use `useCoreKitKey: true` while initializing the Web3Auth Core SDK.

```js
const web3authCore = new Web3AuthCore({
    clientId,
    chainConfig,
    web3AuthNetwork: "cyan",
    useCoreKitKey: true // <-- this is a new addition, please update your web3auth SDK to latest.
});
```

**Comment by chaitanyapotti** [comment]
This is fixed in the latest version. Pls update

**Comment by nisa10880** [comment]
Thanks, it's working now.
